### PR TITLE
Feature/extra features

### PR DESCRIPTION
**Pull request by dcochico**
## What I did:
- Convert delete pop-ups to other modal style
- Allow unplanted cell drag-and-drop

## Issues closed:
#127 
#128 

## Did this break anything?
- [x] No
- [ ] Yes

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ] Super small fix (Corrected a typo, removed a comment, etc.)
- [ ] Skip all the other stuff and briefly explain the fix.

## Checklist:
- [x] If this code needs to be tested, all tests are passing
- [x] The code runs locally
- [ ] There are comments where clarification/ organization is needed
- [ ] The code is DRY. If it's not, I tried my best
- [x] I reviewed my code before pushing

### CODE REVIEW

File: src/components/ConfirmModal/ConfirmModal.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  return (
+    <div className='confirm-modal-overlay' >
+      <div className='confirm-modal' >
+        {fullClear && <p className='confirm-message'>Are you sure you wish to clear your garden? This action cannot be undone.</p>}
+        {!fullClear && <p className='confirm-message'>Are you sure you wish to remove your unplanted items? This action cannot be undone.</p>}
+        <div className='pop-up-button-container'>
+          <button className='pop-up-button' onClick={handleDelete} >YES</button>
+          <button className='pop-up-button' onClick={reset} >NO</button>
```
Comment (mbenfowler) [inline src/components/ConfirmModal/ConfirmModal.tsx:33]: These buttons are little difficult to see at first glance until you hover over them.  Maybe a thicker border? Or make the color and on-hover color opposite?  Not sure hmm.

File: src/components/Grid/Grid.tsx [hunk 2]
```diff
@@ -66,6 +73,7 @@ const Grid = ({ alert, setAlert, garden, setGarden, bullDoze, setBullDoze, filte
   return (
     <section id='grid'>
       {cells}
+      {popUp && <ConfirmModal setPopUp={setPopUp} fullClear={fullClear} setFullClear={setFullClear} setPartialClear={setPartialClear} reset={reset} setBullDoze={setBullDoze} setFilterGarden={setFilterGarden} setAlert={setAlert} />}
       {modal && <Modal alert={alert} bullDoze={bullDoze} filterGarden={filterGarden} toggleModal={toggleModal} />}
```
Comment (mbenfowler) [inline src/components/Grid/Grid.tsx:77]: K i've looked at this for about 30 minutes now and can't determine a cause, but it looks there are a few issues:
1) We lost an alert modal when dragging a Card, and now a Cell, over a `planted` cell
2) If you drag an unplanted plant over a disabled cell (which causes an alert modal), the unplanted plant disappears
3) Dragging a Card, (and now a Cell), over a cell with a non-planted plant doesn't overwrite it anymore - is this intended?

File: src/components/ConfirmModal/ConfirmModal.tsx
Comment (mbenfowler) [inline src/components/ConfirmModal/ConfirmModal.tsx:1]: GREAT work on refactoring this functionality!!

**Review by mbenfowler — CHANGES_REQUESTED** [review]
A few things to look at!

File: src/components/Grid/Grid.tsx (hunk 2 above)
Comment (dcochico) [reply src/components/Grid/Grid.tsx:79]: all of these bugs have been fixed!

File: src/components/Cell/Cell.tsx [hunk 3]
```diff
@@ -227,21 +246,35 @@ const Cell = ({ id, garden, setGarden, bullDoze, setBullDoze, filterGarden, setF
...
+          )}
+          <div
+            id={id}
+            className={[className, wateringWarning].join(' ')}
+            style={{ ...divStyle, ...hoverStyle }}
+            onClick={handleClick}
+            ref={(node) => {
+              dragRef(node)
+              dropRef(node)
+            }}
```
Comment (mbenfowler) [inline src/components/Cell/Cell.tsx:261]: very nice

File: src/components/ConfirmModal/ConfirmModal.scss
Comment (mbenfowler) [inline src/components/ConfirmModal/ConfirmModal.scss:1]: I'm still having an issue with the color scheme on the remove modals:
Screenshot 2023-09-21 at 12 05 01 PM

Can we have them look more like the cell unavailable modal? Or maybe make the top text white? I'd love @AlexaVanSchaardenburg 's thoughts on this too.

File: src/components/Grid/Grid.tsx [hunk 4]
```diff
@@ -15,8 +16,16 @@ export interface GridProps {
...
+  popUp: boolean;
+  setPopUp: Function;
+  fullClear: boolean;
+  setFullClear: Function;
+  setPartialClear: Function;
+  reset: () => void;
   alert: boolean;
   setAlert: Function;
+  modal: boolean;
+  setModal: Function;
```
Comment (mbenfowler) [inline src/components/Grid/Grid.tsx:28]: typez for dayz

**Review by mbenfowler — CHANGES_REQUESTED** [review]
A couple more things to look at - and should we add more cypress tests for new expected Card and Cell drag behavior?

File: src/components/Card/Card.tsx [hunk 5]
```diff
@@ -20,7 +20,7 @@ const Card = ({ plant, modal, setModal }: CombinedCardProps) => {
     type: 'plant',
     item: { plant },
     end: (item, monitor) => {
-      if (!monitor.didDrop()) {
+      if (!monitor.didDrop() && monitor.getTargetIds().length) {
```
Comment (mbenfowler) [inline src/components/Card/Card.tsx:23]: ohhhh yah i think i did see you use this elsewhere, love it!!

File: src/components/ConfirmModal/ConfirmModal.scss [hunk 6]
```diff
@@ -27,17 +27,19 @@
...
-.pop-up-button:hover {
+.pop-up-button {
+  border: none;
+  border-radius: 1em;
   background-color: #af2268;
   color: #f4f4f4;
   cursor: pointer;
+  padding: 0.5em;
+  width: 30%;
+  box-shadow: 0 4px 4px 0 rgba(0, 0, 0, 0.4);
```
Comment (mbenfowler) [inline src/components/ConfirmModal/ConfirmModal.scss:44]: yah this looks way nicer, thank you!

**Review by mbenfowler — APPROVED** [review]
Looks great, we can talk about adding more cypress tests later!

**Pull request merged**